### PR TITLE
CLN: .format to f-string formatting in various files

### PR DESCRIPTION
**Pull request by kpmccahill**
- [ ] ref https://github.com/pandas-dev/pandas/issues/29547
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_config/config.py [hunk 1]
```diff
@@ -461,16 +461,17 @@ def register_option(key: str, defval: object, doc="", validator=None, cb=None):
...
+            option = ".".join(path[:i])
+            raise OptionError(f"Path prefix to option '{option}' is already an option")
         if p not in cursor:
             cursor[p] = {}
         cursor = cursor[p]
 
     if not isinstance(cursor, dict):
-        raise OptionError(msg.format(option=".".join(path[:-1])))
+        option = ".".join(path[:-1])
+        raise OptionError(f"Path prefix to option '{option}' is already an option")
```
Comment (gfyoung) [inline pandas/_config/config.py]: @WillAyd @datapythonista : I'm somewhat ambivalent about this change actually.  As much as I appreciate the convenience of f-strings, the fact we have to duplicate the error message gives me pause.

Thoughts?

**Comment by gfyoung** [comment]
@kpmccahill : Thanks for opening this PR!  And welcome to pandas! 🚀 

The changes look pretty good overall.  I have one question about them that's directed towards fellow maintainers as you can see above.

File: pandas/_config/config.py (hunk 1 above)
Comment (WillAyd) [reply pandas/_config/config.py]: Yea valid point. Probably better use case for .format

**Comment by kpmccahill** [comment]
@gfyoung 
> I have one question about them that's directed towards fellow maintainers as you can see above.

Thanks! And I'm gonna be honest I wasn't too sure about that one myself for the same reason you mentioned. But I thought I'd give it a go anyways just to see.

File: pandas/_config/config.py [hunk 2]
```diff
@@ -388,7 +388,7 @@ class option_context:
     Examples
     --------
 
-    >>> with option_context('display.max_rows', 10, 'display.max_columns', 5):
+    with option_context('display.max_rows', 10, 'display.max_columns', 5):
```
Comment (gfyoung) [inline pandas/_config/config.py]: What happened here?

Comment (kpmccahill) [reply pandas/_config/config.py]: I thought those were some characters I had accidentally added, but now that I look at the master I realize they're supposed to be there, oops

**Review by datapythonista — APPROVED** [review]
lgtm, thanks for the clean up!

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @kpmccahill !

**Pull request merged**